src/cpp/shared/town/town.cpp

```diff
@@ -1210,9 +1210,10 @@ void townManager::SetupCastle(heroWindow *window, int a3) {
 
   if(imgIdx == -1) {
     GUIRemoveFlag(casWin, 1102, 4);
-    GUISetImgIdx(casWin, 1102, imgIdx);
-  } else
-    GUIAddFlag(casWin, 1102, 4);
+  } else {
+	GUISetImgIdx(casWin, 1102, imgIdx);
```

Comment:
@Kert  Can you explain how this code works?

Reply:
It's a typo. Previously it was setting imgIdx to a widget when it was -1, which is not right obviously